### ISSUE TITLE
[Bug]: Font is not smooth in Notepad

### ISSUE THREAD

**Issue by ertankucukoglu** [MonoLisaFont/feedback#281]
### Edition

Plus

### What version are you using?

2.015

### What type of weights are you using?

Default (non-customized)

### Operating system

Windows 11

### Program

Notepad

### Display resolution

2560 x 1440 @ 100% scale

### What happened?

This is MonoLisa Regular, I have ss02 for scripts but there is no script in screen shots below.

I see font rendering is not smooth for especially 11 pixels. Letters G, T, P it is a little more obvious. 12px it gets better, but this time [ character seems to be thicker at top.

I tried other programs I use and it is only Notepad that font is not smooth.

I don't know if this is normal so just wanted to let you know.

10px
![image](https://github.com/user-attachments/assets/f10ee60a-6054-4cae-8267-b6c47c4078f8)

11px
![image](https://github.com/user-attachments/assets/f9b5195b-0a40-4270-9584-64d0cfabdcf3)

12px
![image](https://github.com/user-attachments/assets/c426ff7f-218c-451a-9b1a-989bec68292b)

14px
![image](https://github.com/user-attachments/assets/10d803f8-319e-4959-8370-9b7420e2f8e2)

Thanks & Regards,
Ertan

**Comment by bebraw** [comment]
Out of curiosity, do you see the issue in other editors like VS Code?

**Comment by ertankucukoglu** [comment]
No, as I told problem seems to be only in Notepad. (Windows Notepad 11.2407.9.0)
I don't use VS Code anyway, but anything I use the MonoLisa is OK for me.

**Comment by bebraw** [comment]
Gotcha, maybe it's using a different rendering path somehow.

Can you see the same behavior with any other programming font in Notepad?

**Comment by ertankucukoglu** [comment]
I do not see that problem with some of the other fonts I tried.

IBM Plex Mono 10px
![image](https://github.com/user-attachments/assets/91664e07-6d2a-4929-a15f-548424e4a334)

IBM Plex Mono 11px
![image](https://github.com/user-attachments/assets/d8e23bde-36a0-49c5-a8c1-3bf382d1d021)

IBM Plex Mono 12px
![image](https://github.com/user-attachments/assets/9298beb5-3f2a-4a5c-aa2f-14ae6f5da4f2)

Fira Code 10px
![image](https://github.com/user-attachments/assets/8b135809-0127-4c03-8d4a-82a108e3157d)

Fira Code 11px
![image](https://github.com/user-attachments/assets/3b0fb6b9-2a8b-4712-bd44-92faa476f514)

Fira Code 12px
![image](https://github.com/user-attachments/assets/ffab6a49-5f87-408e-8bd7-6e99fd0992ac)

Consolas 10px
![image](https://github.com/user-attachments/assets/7a4fc8d6-c720-4b5e-8143-d06136992b3d)

Consolas 11px
![image](https://github.com/user-attachments/assets/08386b71-09e2-4142-8bdc-3f0fea106f69)

Consolas 12px
![image](https://github.com/user-attachments/assets/a1b1681f-634c-4598-9437-115ff1780cdf)

**Comment by bebraw** [comment]
Thanks, that's good info and will help in debugging.

**Comment by spapaseit** [comment]
> Out of curiosity, do you see the issue in other editors like VS Code?

Just to add my two cents, I do see the same problem in Visual Studio 2022 (ARM, in case it matters), only at 100% zoom level weirdly enough

100% zoom (font size 13):
![image](https://github.com/user-attachments/assets/46682f2a-7cf9-4fef-ba80-c84b4a858a79)

But at 102% zoom, the font is crisp:
![image](https://github.com/user-attachments/assets/9b7f9ce7-b12f-4cab-b2fd-4496d3192801)

And so it is at 90%:
![image](https://github.com/user-attachments/assets/4ae403d1-5841-4df3-b33a-9982d3a4c89a)

Font looks perfect in VS Code
I'm using MonoLisa v2.015 (purchased and installed yesterday - beautiful font!).

**Comment by bebraw** [comment]
Related commits from Fira Code:

* https://github.com/tonsky/FiraCode/commit/bd4e557891b901a89ad1b088a5e10f916a7dac11#diff-d5ab68cc52acb21a2a41b721082a3357dfd3b66b6002afffa6984c8f5cf341eaR23
* https://github.com/tonsky/FiraCode/commit/8da49d55f8b5978c5f888dd85452b79aad16cca2
* https://github.com/tonsky/FiraCode/commit/31adb247e53ff2f58dc4136e61feac806e84b9de

The interesting point is that Fira Code uses `ttfautohint` for applying some kind of fixes.

**Comment by bebraw** [comment]
Wrappers: https://github.com/fonttools/ttfautohint-py, https://www.npmjs.com/package/ttfautohint .

@ertankucukoglu If you want, try `ttfautohint --no-info --ignore-restrictions ${file} ${file}.hinted` kind of command against the font files to see if that would help with hinting. Thanks!

**Comment by ertankucukoglu** [comment]
I am not sure I did it correctly. Here is the generated file.

**Comment by bebraw** [comment]
@ertankucukoglu If you change the generated filename to plain `.ttf`, you should be able to test the hinted version locally on your machine. This might need a restart due to font caching (hopefully you know how to purge font cache for Windows).

Note that I dropped the zip from your comment after checking as we don't have to share it in public. :)

**Comment by bebraw** [comment]
@ertankucukoglu Did the font look better for you with the autohinted version?

**Comment by ertankucukoglu** [comment]
@bebraw I am sorry. I don't know what is "autohinted version".

**Comment by bebraw** [comment]
@ertankucukoglu I mean the version you generated using `ttfautohint`.

**Comment by ertankucukoglu** [comment]
I installed over writing the existing Regular font. Seems fine to me.
10px
![image](https://github.com/user-attachments/assets/7ad70ebc-2dba-4c0e-bd48-9708a56fdea2)

11px
![image](https://github.com/user-attachments/assets/37c18594-39a8-4da6-b0ec-cec889126416)

12px
![image](https://github.com/user-attachments/assets/a3e2f901-ed0d-469f-977e-7b8cc2229825)

**Comment by bebraw** [comment]
Thanks for the info. That's good news and a nice workaround for now. Let's leave the issue open as we might address this somehow.

**Comment by ertankucukoglu** [comment]
I might have spoken early. I just realize some small letters "p", "u" are not inline with others but positioned a little below.
![image](https://github.com/user-attachments/assets/9e570276-0dd8-4dc9-b63d-4729f0345fa0)

This happens on all editors that I use

**Comment by bebraw** [comment]
@ertankucukoglu Yeah, that's an interesting artefact. There might be some better way to run the autohinting tool. From [their documentation](https://freetype.org/ttfautohint/doc/ttfautohint.html) I noticed they have a Windows specific flag (`--windows-compatibility`) which might help.

**Comment by ertankucukoglu** [comment]
That parameter didn't help much. There is still that artifact of having some letters not aligned same with the others.
![image](https://github.com/user-attachments/assets/b248a0bd-0ad0-454b-94b2-70d94831eb75)

**Comment by bebraw** [comment]
Gotcha, thanks for checking!
